src/hal/AngleSensor.h

```diff
@@ -3,12 +3,16 @@
 
 #include "SensorInterface.h"
 
-class AngleSensor : public SensorInterface {
+// int is just a placeholder
+class AngleSensor : public SensorInterface<double, int> {
 public:
     explicit AngleSensor(double initialAngle);
-    double GetAngle() override;
+    double GetValue(int gpio) override;
```

Comment:
We should tell the compiler that every time we call the method, we MUST use the data that comes from it.
This can be done by [attributes](https://en.cppreference.com/w/cpp/language/attributes), more specifically [nodiscard](https://en.cppreference.com/w/cpp/language/attributes/nodiscard), so:

```cpp
[[nodiscard]] double GetValue() override;
```

If we don't use the value (assigning to a variable or whatever) the compiler will throw an error.

Reply:
Cool! I've never really used attribute specifiers, I'll study them a bit!